### PR TITLE
ci: pin rust version in CI

### PR DESCRIPTION
**Pull request by tychoish**
Closes: https://github.com/GlareDB/glaredb/issues/1648

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -20,7 +20,8 @@ jobs:
         uses: actions/checkout@v4
 
       - name: Rust toolchain
-        uses: dtolnay/rust-toolchain@stable
+        uses: dtolnay/rust-toolchain@1.72.0
+        components: clippy,rstfmt
```
Comment (scsmithr) [inline .github/workflows/ci.yaml:24]: This needs to be:
```
with:
  components: clippy,rustfmt
```

Comment (tychoish) [reply .github/workflows/ci.yaml:24]:
```suggestion
        components: clippy,rustfmt
```

Comment (tychoish) [reply .github/workflows/ci.yaml]: The thing that is confusing is how this passed at all.

Comment (tychoish) [reply .github/workflows/ci.yaml]: > This actual "Build and Test" workflow isn't running,

ah that's confusing that syntax failures in the yaml file don't result in failed tasks?

thanks for the eyeballs: I would have actually looked if the task had had something that wasn't a check mark next to it?

**Review by scsmithr — APPROVED** [review]

**Pull request merged**